--- a/src/bsa/fo4.cpp
+++ b/src/bsa/fo4.cpp
@@ -740,7 +740,7 @@
 		const DirectX::TexMetadata meta{
 			.width = this->header.width,
 			.height = this->header.height,
 			.depth = 1,
-			.arraySize = isCubemap ? 6 : 1,
+			.arraySize = static_cast<std::size_t>(isCubemap ? 6 : 1),
 			.mipLevels = this->header.mip_count,
 			.miscFlags = isCubemap ? std::uint32_t{ DirectX::TEX_MISC_FLAG::TEX_MISC_TEXTURECUBE } : 0u,